src/components/Modal/BaseModal.js

```diff
@@ -51,8 +51,10 @@ class BaseModal extends PureComponent {
     }
 
     componentWillUnmount() {
-        // we don't want to call the onModalHide on unmount
-        this.hideModal(this.props.isVisible);
+        // Only trigger onClose and setModalVisibility if the modal is unmounting while visible.
+        if (this.props.isVisible) {
+            this.hideModal(true);
+        }
 
         if (this.props.isVisible) {
             Modal.willAlertModalBecomeVisible(false);
```

Comment:
```suggestion
            this.hideModal(true);
            Modal.willAlertModalBecomeVisible(false);
```

Just realized we're making `this.props.isVisible` check twice, we can move `this.hideModal(true) `here